### PR TITLE
Start VM with custom kernel

### PR DESCRIPTION
**Pull request by WeiZhang555**
Now runv-containerd support embedded kernel in image, if one custom
kernel/initrd exists in container image, use it first instead of the
system default kernel/initrd files.

Signed-off-by: Zhang Wei <zhangwei555@huawei.com>

### CODE REVIEW

File: supervisor/hyperpod.go [hunk 1]
```diff
@@ -77,10 +78,51 @@ func createHyperPod(f factory.Factory, spec *specs.Spec) (*HyperPod, error) {
...
-	if err != nil {
-		glog.V(1).Infof("%s\n", err.Error())
-		return nil, err
+
+	// if image contains kernel file then use it first instead of default kernel
+	kernel := utils.FirstExistingFile([]string{
+		filepath.Join(spec.Root.Path, "boot/vmlinuz"),
+		filepath.Join(spec.Root.Path, "vmlinuz"),
+		filepath.Join(spec.Root.Path, "kernel"),
+		hypervisor.DefaultKernel,
```
Comment (laijs) [inline supervisor/hyperpod.go]: please remove hypervisor.DefaultKernel.
use `kernel == ""` insteead of `kernel == hypervisor.DefaultKernel`

Comment (WeiZhang555) [inline supervisor/hyperpod.go]: Then how to guarantee "/var/lib/hyper" exists? I've tried the latest master, if default kernel doesn't exist, runv will still try to boot the VM. 
With this change, we can guarantee hypervisor.DefaultKernel really exists

**Comment by laijs** [comment]
any update?

**Comment by WeiZhang555** [comment]
What's our new plan? 
Keep kernel file in image or pass kernel parameter with ENV (-e runv.kernel=xxxx)?

**Comment by laijs** [comment]
-e hypervisor.kernel=xxxx

**Comment by WeiZhang555** [comment]
OK. Busy there days, I'll update once get time :smile:

**Comment by WeiZhang555** [comment]
@laijs I've already updated this PR. 
Now it can support `-e hypervisor.kernel` and `-e hypervisor.initrd`.
Also, it still support embedded kernel+initrd in docker image.
PTAL.

**Comment by WeiZhang555** [comment]
How about we forbid embedded kernel file in image? After some consideration, I think it's a little bit dangerous, since we can't control source of the image but we can control how to start the container with environment.

File: supervisor/hyperpod.go [hunk 2]
```diff
@@ -304,10 +345,40 @@ func createHyperPod(f factory.Factory, spec *specs.Spec) (*HyperPod, error) {
...
+			glog.V(1).Infof("Create VM failed with default kernel config: %s", err.Error())
+			return nil, err
+		}
+	} else if len(kernel) == 0 || len(initrd) == 0 {
+		// if user specify a kernel, they must specify an initrd at the same time
+		return nil, fmt.Errorf("You must specify an initrd if you specify a kernel, or vice-versa")
+	} else {
+		if len(kernel) == 0 {
+
+		}
```
Comment (laijs) [inline supervisor/hyperpod.go]: there is empty code block here.

Comment (WeiZhang555) [inline supervisor/hyperpod.go]: Oops, sorry. Miss this.

**Comment by WeiZhang555** [comment]
Updated.

**Pull request merged**